bundle-workflow/python/paths/script_finder.py

```diff
@@ -0,0 +1,43 @@
+# Copyright OpenSearch Contributors.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from dataclasses import dataclass
+
+@dataclass
+class ScriptFinder:
+    '''
+    ScriptFinder is a helper that abstracts away the details of where to look for build and test scripts.
+    Given a component name and a checked-out Git repository, it will look in the following locations, in order, for either "build.sh" or "integtest.sh", depending on the use case:
+      * Root of the Git repository
+      * /scripts/<script-name> in the Git repository
+      * <component_scripts_path>/<component_name>/<script-name>
+      * <default_scripts_path>/<script-name>
+    '''
+
+    component_scripts_path: str
+    default_scripts_path: str
+
+    def find_build_script(self, component_name, git_dir):
```

Comment:
Notice that both `find_build_script` and `find_integ_test_script` are identical minus the name of the script, `build.sh`, so these methods can combine in one common implementation.

Reply:
I considered that, but decided that the client api should have two methods rather than asking the client to know the name of the script to fetch. That way, typos will result in a NoMethodError (or whatever it is that Python throws when it doesn't recognize a method name) rather than a RuntimeError saying it can't find `bui1d.sh`.

Internally it could totally be refactored into a private helper. Future PR!